--- a/src/libgit2/repository.c
+++ b/src/libgit2/repository.c
@@ -567,16 +567,22 @@
 	} else if (strcmp(entry->value, "*") == 0) {
 		*data->is_safe = true;
 	} else {
+		bool is_prefix = false, match;
 		if (git_str_sets(&data->tmp, entry->value) < 0)
 			return -1;
+
+		if (git__suffixcmp(data->tmp.ptr, "/*") == 0) {
+			is_prefix = true;
+			git_str_shorten(&data->tmp, 1);
+		}
 
 		if (!git_fs_path_is_root(data->tmp.ptr)) {
 			/* Input must not have trailing backslash. */
 			if (!data->tmp.size ||
-			    data->tmp.ptr[data->tmp.size - 1] == '/')
+			    (!is_prefix && data->tmp.ptr[data->tmp.size - 1] == '/'))
 				return 0;
 
-			if (git_fs_path_to_dir(&data->tmp) < 0)
+			if (!is_prefix && git_fs_path_to_dir(&data->tmp) < 0)
 				return -1;
 		}
 
@@ -608,7 +614,10 @@
 		if (strncmp(test_path, "%(prefix)//", strlen("%(prefix)//")) == 0)
 			test_path += strlen("%(prefix)/");
 
-		if (strcmp(test_path, data->repo_path) == 0)
+		match = is_prefix ?
+			(git__prefixcmp(data->repo_path, test_path) == 0) :
+			(strcmp(test_path, data->repo_path) == 0);
+		if (match)
 			*data->is_safe = true;
 	}
 
